asserts/model.go

```diff
@@ -666,31 +674,15 @@ func checkAuthorityMatchesBrand(a Assertion) error {
 	return nil
 }
 
-func checkOptionalSerialAuthority(headers map[string]interface{}, brandID string) ([]string, error) {
-	ids := []string{brandID}
-	const name = "serial-authority"
-	if _, ok := headers[name]; !ok {
-		return ids, nil
-	}
-	if lst, err := checkStringListMatches(headers, name, validAccountID); err == nil {
-		if !strutil.ListContains(lst, brandID) {
-			lst = append(ids, lst...)
-		}
-		return lst, nil
-	}
-	return nil, fmt.Errorf("%q header must be a list of account ids", name)
-}
-
-func checkOptionalSystemUserAuthority(headers map[string]interface{}, brandID string) ([]string, error) {
+func checkOptionalAuthority(name string, acceptsAny bool, headers map[string]interface{}, brandID string) ([]string, error) {
```

Comment:
This is a really nice genericization :) One nitpicky recommendation: passing bools into functions as a standalone param is not so nice from a readability standpoint. More readable might be defining a struct like (except formatted correctly) 
```
type authDelegationMetaParams struct {
    key string
    acceptsAny bool
}
```
(where `name` -> `key` as I think this is not describing the name of an assert, but simply a key in the model)

and then the signature can be `func checkOptionalAuthority(keyParams authDelegationMetaParams, headers...)`. Conceptually, I divided the metaparams from the other two, because those are the two that probably will be build-time static, and the other two are used at runtime (although that should make little difference to the compiler)

Reply:
that's generally true but this is a helper that gets reencapsulated in other more specifc helpers with one just one bool, what I would recommend is:

```
func checkOptionalAuthority(headers map[string]interface{}, name string, brandID string, acceptsAny bool) ...
```

for consistency with other check* helpers, and then in the wrappers:

```
func checkOptionalFoor(headers map[string]interface{}, brandID string) ([]string, error) {
        const acceptsAny = true|false;
	return checkOptionalAuthority(headers, "foo", brandID, acceptsAny)
}
```

with the const used for clarity